From b8ba23b222923c7f1d03c5b3b061938860ca7441 Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Sun, 21 Apr 2013 13:06:22 +0200
Subject: [PATCH 333/364] 	* grub-core/disk/arc/arcdisk.c
 (grub_arcdisk_iterate_iter): 	Fix a type which prevented CD-ROM and floppy
 boot.

---
 ChangeLog                    | 5 +++++
 grub-core/disk/arc/arcdisk.c | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 6be583e..2150d3d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2013-04-21  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* grub-core/disk/arc/arcdisk.c (grub_arcdisk_iterate_iter):
+	Fix a type which prevented CD-ROM and floppy boot.
+
+2013-04-21  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	Support coreboot framebuffer.
 
 	* grub-core/video/i386/coreboot/cbfb.c: New file.
diff --git a/grub-core/disk/arc/arcdisk.c b/grub-core/disk/arc/arcdisk.c
index 5d12128..780728f 100644
--- a/grub-core/disk/arc/arcdisk.c
+++ b/grub-core/disk/arc/arcdisk.c
@@ -95,7 +95,7 @@ grub_arcdisk_iterate_iter (const char *name,
   struct grub_arcdisk_iterate_ctx *ctx = data;
 
   if (!(comp->type == GRUB_ARC_COMPONENT_TYPE_DISK
-	|| comp->type == GRUB_ARC_COMPONENT_TYPE_DISK
+	|| comp->type == GRUB_ARC_COMPONENT_TYPE_FLOPPY
 	|| comp->type == GRUB_ARC_COMPONENT_TYPE_TAPE))
     return 0;
   return ctx->hook (name, ctx->hook_data);
-- 
1.8.1.4

